### PR TITLE
update debugger code for ws url changes

### PR DESCRIPTION
**Pull request by devoncarew**
- update debugger code for ws url changes
- fix https://github.com/flutter/flutter-intellij/issues/453

@pq

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,5 +1,6 @@
 out/
 .gradle/
```
Comment (pq) [inline .gitignore]: Aside: do we need this anymore?

File: src/io/flutter/actions/OpenObservatoryAction.java [hunk 2]
```diff
@@ -20,6 +20,16 @@
 
 @SuppressWarnings("ComponentNotRegistered")
 public class OpenObservatoryAction extends DumbAwareAction {
+  public static String convertWsToHttp(String wsUrl) {
```
Comment (pq) [inline src/io/flutter/actions/OpenObservatoryAction.java]: Can this be `null`?

File: src/io/flutter/run/FlutterRunner.java [hunk 3]
```diff
@@ -79,8 +79,8 @@ public boolean isConnectionReady() {
         }
 
         @Override
-        public int getPort() {
-          return appState.getObservatoryPort();
+        public String getObservatoryWsUrl() {
```
Comment (pq) [inline src/io/flutter/run/FlutterRunner.java]: Maybe annotate...  `@Nullable`?

File: src/io/flutter/run/daemon/FlutterApp.java [hunk 4]
```diff
@@ -89,14 +89,14 @@ public Project project() {
...
    */
-  public int port() {
-    return myPort;
+  public String wsUrl() {
+    return myWsUrl;
   }
 
-  public void setPort(int port) {
-    myPort = port;
+  public void setWsUrl(String url) {
```
Comment (pq) [inline src/io/flutter/run/daemon/FlutterApp.java]: Can url be `null`?

File: src/io/flutter/run/daemon/FlutterApp.java [hunk 5]
```diff
@@ -89,14 +89,14 @@ public Project project() {
   }
 
   /**
-   * @return The debug port used to talk to the observatory.
+   * @return the Observatory WebSocket URL
    */
-  public int port() {
-    return myPort;
+  public String wsUrl() {
```
Comment (pq) [inline src/io/flutter/run/daemon/FlutterApp.java]: I'm guessing this is `@Nullable`?

File: third_party/intellij-plugins-dart/src/com/jetbrains/lang/dart/ide/runner/ObservatoryConnector.java [hunk 6]
```diff
@@ -11,9 +11,9 @@
   boolean isConnectionReady();
 
   /**
-   * Return the port used by the observatory.
+   * Return the WebSocket URL used by the observatory.
    */
-  int getPort();
+  String getObservatoryWsUrl();
```
Comment (pq) [inline third_party/intellij-plugins-dart/src/com/jetbrains/lang/dart/ide/runner/ObservatoryConnector.java]: The old saw: `@Nullable`?

File: .gitignore (hunk 1 above)
Comment (devoncarew) [inline .gitignore]: no, will delete

File: src/io/flutter/actions/OpenObservatoryAction.java (hunk 2 above)
Comment (devoncarew) [inline src/io/flutter/actions/OpenObservatoryAction.java]: added an annotation

File: src/io/flutter/run/FlutterRunner.java (hunk 3 above)
Comment (devoncarew) [inline src/io/flutter/run/FlutterRunner.java]: kk

File: src/io/flutter/run/daemon/FlutterApp.java (hunk 5 above)
Comment (devoncarew) [inline src/io/flutter/run/daemon/FlutterApp.java]: yup, added

File: src/io/flutter/run/daemon/FlutterApp.java (hunk 4 above)
Comment (devoncarew) [inline src/io/flutter/run/daemon/FlutterApp.java]: no, added

File: third_party/intellij-plugins-dart/src/com/jetbrains/lang/dart/ide/runner/ObservatoryConnector.java (hunk 6 above)
Comment (devoncarew) [inline third_party/intellij-plugins-dart/src/com/jetbrains/lang/dart/ide/runner/ObservatoryConnector.java]: added :)

**Pull request merged**